res/app/device-list/details/device-list-details-directive.js

```diff
@@ -277,6 +279,20 @@ module.exports = function DeviceListDetailsDirective(
         }
       })()
 
+      // check if new column needs a new scope or not
+      // and build accordingly
+      function buildColumn(col, device) {
```

Comment:
@gunta  @sorccu 

I fixed performance issue by adding new scope only when required. I tested with around 1000 fake device, reload time is kind of same for this and master branch.

I also changed design by tweaking angular-xeditable css to make it popover. I have attached new screenshots in description. Have a look.

If you think there are other better way to do this, please suggest.